dipy/workflows/align.py

```diff
@@ -45,16 +54,492 @@ def run(self, input_files, new_vox_size, order=1, mode='constant', cval=0,
+            image. The default is None, implying the transform is the
+            identity.
+
+        moving : array, shape (S', R', C') or (R', C')
+            the image to be used as "moving" during optimization. It is
+            necessary to pre-align the moving image to ensure its domain
+            lies inside the domain of the deformation fields. This is assumed
+            to be accomplished by "pre-aligning" the moving image towards the
+            static using an affine transformation given by the
+            'starting_affine' matrix
+
+        moving_grid2world : array, shape (dim+1, dim+1), optional
+            the voxel-to-space transformation associated with the moving
+            image. The default is None, implying the transform is the
+            identity.
+
+        """
+
+        img_registration = transform_centers_of_mass(static,
+                                                     static_grid2world,
+                                                     moving,
+                                                     moving_grid2world)
+
+        transformed = img_registration.transform(moving)
+        return transformed, img_registration.affine
+
+    def translate(self, static, static_grid2world, moving,
+                  moving_grid2world, affreg, params0):
+
+        """ Function for translation based registration.
+
+        Parameters
+        ----------
+        static : array, shape (S, R, C) or (R, C)
+            the image to be used as reference during optimization.
+
+        static_grid2world : array, shape (dim+1, dim+1), optional
+            the voxel-to-space transformation associated with the static
+            image. The default is None, implying the transform is the
+            identity.
+
+        moving : array, shape (S', R', C') or (R', C')
+            the image to be used as "moving" during optimization. It is
+            necessary to pre-align the moving image to ensure its domain
+            lies inside the domain of the deformation fields. This is assumed
+            to be accomplished by "pre-aligning" the moving image towards the
+            static using an affine transformation given by the
+            'starting_affine' matrix
+
+        moving_grid2world : array, shape (dim+1, dim+1), optional
+            the voxel-to-space transformation associated with the moving
+            image. The default is None, implying the transform is the
+            identity.
+
+        affreg : An object of the image registration class.
+
+        params0 : array, shape (n,)
+            parameters from which to start the optimization. If None, the
+            optimization will start at the identity transform. n is the
+            number of parameters of the specified transformation.
+
+        """
+        moved, affine = self.center_of_mass(static, static_grid2world,
+                                            moving, moving_grid2world)
```

Comment:
Looks like you're not using `moved` here and below. Simply use `_, affine = ...`

Reply:
Done. I forgot to take advantage of this feature of Python :) Thanks for reminding me.